.github/workflows/check_interface_compatibility.yml

```diff
@@ -0,0 +1,80 @@
+name: Check interface compatibility
+
+on:
+  pull_request_target:
+
+permissions:
+  pull-requests: write
+
+jobs:
+  check-interface-compatibility:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Extract original public interfaces
+        run: |
+          make init
+          bin/public_interface.py extract > ${{ runner.temp }}/interfaces.original.json
+
+      - name: Backup original bin/public_interface.py
+        run: |
+          # Keep a copy of bin/public_interface.py to use on the base branch
+          # So we are using the same bin/public_interface.py to process old/new codebase.
+          # Because the step has credential of GitHub token to post comments,
+          # always use the original version that is in our repository.
+          cp bin/public_interface.py ${{ runner.temp }}/public_interface.py
+
+      - name: Checkout the PR
+        uses: actions/checkout@v3
+        with:
+          ref: "refs/pull/${{ github.event.number }}/merge"
+
+      - name: Extract PR public interfaces
+        run: |
+          make init
+          # Recover bin/public_interface.py
+          cp ${{ runner.temp }}/public_interface.py bin/public_interface.py
+          bin/public_interface.py extract > ${{ runner.temp }}/interfaces.new.json
+    
+      - name: Detect compatibility breaking changes
+        id: detect
+        run: |
+          exit_code=0
+          output=$(bin/public_interface.py check ${{ runner.temp }}/interfaces.original.json ${{ runner.temp }}/interfaces.new.json) || exit_code=$?
```

Comment:
AFAIK you don't need the initialization or `||` thing. Just `exit_code=$?` on separate line should be enough.

Reply:
same issue as we saw in the release script, if `bin/public_interface.py` exit 1, the whole job stops without `||`